epflldap/ldap_search.py

```diff
@@ -64,10 +68,19 @@ def get_unit_id(unit_name):
     """
     attribute = 'uniqueIdentifier'
     response = LDAP_search(
-        pattern_search='(cn=' + unit_name + ')',
+        pattern_search='(cn={})'.format(unit_name),
         attribute=attribute
     )
-    return response[0]['attributes'][attribute][0]
+
+    unit_id = ""
+    for element in response:
+        if 'dn' in element and element['dn'].startswith('ou={},'.format(unit_name)):
+            unit_id = element['attributes'][attribute][0]
```

Comment:
à traiter de la même façon que les autres (c'est à dire avec un try except 🐎  pour l'instant) 😬

Reply:
pour cette méthode, je t'accorde que c'est un peu arbitraire d'avoir choisi de ne pas mettre en try except en mode bourrin mais je n'avais pas envie de répéter 2 fois 
raise EpflLdapException("The unit named '{}' was not found".format(unit_name))
et le code me semble assez robuste.
Maisok pour plus de robustesse justement, je vais passer en mode bourrin :-)

Reply:
on peut aussi faire joli et robuste en factorisant ces appels. Ne pas catcher va nous renvoyer des IndexError ou KeyError à un moment ou à un autre (cf mon premier commentaire)